network/nodes.py

```diff
@@ -178,14 +179,16 @@ def _pop_next_api_client(self):
 	def _update(self, public_key, json_node, json_peers):
 		self.public_key_to_node_info_map[public_key] = json_node
 		for json_peer in json_peers:
-			peer_api_client = self.api_client_class.from_node_info_dict(
-				json_peer,
-				retry_count=2,
-				timeout=self.timeout,
-				certificate_directory=self.certificate_directory)
-
-			if peer_api_client and peer_api_client.node_host not in self.visited_hosts:
-				if not any(peer_api_client.node_host == api_client.node_host for api_client in self.remaining_api_clients):
+			host = json_peer.get('host')
+
+			if host not in self.visited_hosts:
+				if not any(host == api_client.node_host for api_client in self.remaining_api_clients):
```

Comment:
These checks might be fine for now, but I think this can be a bottleneck as the network grows since it's a single.

Maybe sometime in the future PR to move the creation of the API client to the thread it is used in....Just a thought
cc: @Jaguar0625